src/components/application_manager/test/app_launch/app_launch_data_db_test.cc

```diff
@@ -0,0 +1,334 @@
+  EXPECT_TRUE(res_db()->AddApplicationData(data));
+
+  uint32_t sizeAfterAdding = res_db()->GetCurentNumberOfAppData();
+  EXPECT_EQ(sizeBeforeAdding, sizeAfterAdding);
+}
+
+std::string AppLaunchDataDBTest::GetApplicationData(
+    const ApplicationData& in_data, ApplicationData& out_data) {
+  utils::dbms::SQLQuery query(test_db());
+  uint32_t sizeBeforeAdding = res_db()->GetCurentNumberOfAppData();
+  EXPECT_TRUE(query.Prepare(kGetSsession));
+  query.Bind(AppLaunchDataDB::device_mac_index, in_data.device_mac_);
+  query.Bind(AppLaunchDataDB::application_id_index, in_data.mobile_app_id_);
+  query.Bind(AppLaunchDataDB::bundle_id_index, in_data.bundle_id_);
+  EXPECT_TRUE(query.Exec());
+
+  uint32_t sizeAfterAdding = res_db()->GetCurentNumberOfAppData();
+
+  EXPECT_EQ(sizeBeforeAdding, sizeAfterAdding);
+
+  out_data.device_mac_ = query.GetString(AppLaunchDataDB::device_mac_index);
+  out_data.mobile_app_id_ =
+      query.GetString(AppLaunchDataDB::application_id_index);
+  out_data.bundle_id_ = query.GetString(AppLaunchDataDB::bundle_id_index);
+  // Timestamp
+  return query.GetString(AppLaunchDataDB::timestamp_index);
+}
+
+std::string AppLaunchDataDBTest::AddCounter(const std::string& inp,
+                                            int32_t val) {
+  std::stringstream ss;
+  ss << inp << val;
+  return ss.str();
+}
+
+TEST_F(AppLaunchDataDBTest, Init) {
+  utils::dbms::SQLQuery query_checks(test_db());
+  EXPECT_TRUE(query_checks.Prepare(kCheckTablesExist));
+  EXPECT_TRUE(query_checks.Exec());
+  EXPECT_EQ(1u, query_checks.GetUInteger(0));
+}
+
+TEST_F(AppLaunchDataDBTest, SaveAndGetData) {
+  ApplicationData data("mobile_app_id", "bundle_id", "device_mac");
+  AddApplicationDataWithIncreaseTable(data);
+  ApplicationData recoveredData("", "", "");
+  GetApplicationData(data, recoveredData);
+  EXPECT_TRUE(data == recoveredData);
+}
+
+TEST_F(AppLaunchDataDBTest, NotAddEmptyAppData) {
+  ApplicationData data1("", "bundle_id", "device_mac");
+  ApplicationData data2("mobile_app_id", "", "device_mac");
+  ApplicationData data3("mobile_app_id", "bundle_id", "");
+  ApplicationData data4("", "", "");
+  EXPECT_FALSE(res_db()->AddApplicationData(data1));
+  EXPECT_FALSE(res_db()->AddApplicationData(data2));
+  EXPECT_FALSE(res_db()->AddApplicationData(data3));
+  EXPECT_FALSE(res_db()->AddApplicationData(data4));
+  EXPECT_EQ(0u, res_db()->GetCurentNumberOfAppData());
+}
+
+TEST_F(AppLaunchDataDBTest, SaveOneAndGetAnotherData) {
+  ApplicationData data("mobile_app_id", "bundle_id", "device_mac");
+  ApplicationData recoverData = data;
+  AddApplicationDataWithIncreaseTable(data);
+  recoverData.device_mac_ += "test";
+  GetApplicationData(recoverData, recoverData);
+  EXPECT_FALSE(data == recoverData);
+}
+
+TEST_F(AppLaunchDataDBTest, MaxCount) {
+  const uint32_t mac_ios_devs = res_db()->get_max_number_iOS_devs();
+  for (uint32_t i = 0; i < mac_ios_devs; i++) {
+    ApplicationData data(AddCounter("mobile_app_id_", i),
```

Comment:
@Kozoriz, in this case it's unnecessary. But other strings, which repeats more  frequently has been replaced by constants in 	ec61caa.